src/components/Table/nestedCards.test.tsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import { NestedCardStyle } from "src/components/Table/GridTable";
+import { makeOpenOrCloseCard, maybeAddCardPadding } from "src/components/Table/nestedCards";
+import { Palette } from "src/Css";
+import { render } from "src/utils/rtl";
+
+const parentCardStyle: NestedCardStyle = {
+  bgColor: Palette.Gray100,
+  brPx: 8,
+  pxPx: 8,
+  spacerPx: 8,
+};
+
+const childCardStyle: NestedCardStyle = {
+  bgColor: Palette.Gray300,
+  bColor: Palette.Gray200,
+  brPx: 6,
+  pxPx: 6,
+  spacerPx: 6,
+};
+
+describe("nestedCards", () => {
+  it("can make open card w/one level", async () => {
+    const r = await render(makeOpenOrCloseCard([parentCardStyle], "open"));
+    expect(r.firstElement).toMatchInlineSnapshot(`
+      .emotion-0 {
+        background-color: rgba(247,245,245,1);
+        padding-left: 8px;
+        padding-right: 8px;
+        border-top-left-radius: 8px;
+        border-top-right-radius: 8px;
+        height: 8px;
+      }
+
+      <div
+        data-overlay-container="true"
+      >
+        <div
+          class="emotion-0"
+        />
+      </div>
+    `);
+  });
+
+  it("can make open card w/two levels", async () => {
+    const r = await render(makeOpenOrCloseCard([parentCardStyle, childCardStyle], "open"));
+    expect(r.firstElement).toMatchInlineSnapshot(`
+      .emotion-0 {
+        background-color: rgba(247,245,245,1);
+        padding-left: 8px;
```

Comment:
**question:** I was thinking that this should also have the border-top left and right too for the parents opening row style?

Reply:
This is drawing a 2nd-level row open, so something like:

```
/--------------------------\
| 1st-level card        |
|  /--------------------\   | <--- this row is what this test is drawing
|  |  2nd-level       |  |
```

So the parent's border-top left/right are above us, and for this row we just straight lines for the parent.